src/core/lib/promise/party.h

```diff
@@ -634,6 +632,7 @@ class Party : public Activity, private Wakeable {
   // If the lower bit is unset, then this is a Participant*.
   // If the lower bit is set, then this is a ParticipantFactory*.
   std::atomic<Participant*> participants_[party_detail::kMaxParticipants] = {};
+  RefCountedPtr<Arena> arena_;
```

Comment:
It's a little unclear at the minute (especially since resource quota depends on parts of promise too). Ack on the problem, let's iterate away `Activity` at least and then see where we stand?

Reply:
Sounds good.  For now, I'll hold off on moving the promise code, and we can figure this out as we go.